lightblue/src/main/java/org/esbtools/eventhandler/lightblue/client/BulkLightblueRequester.java

```diff
@@ -349,20 +353,60 @@ public boolean isCancelled() {
         public boolean isDone() {
             return cancelled || completed;
         }
+        
+        private static class TimeoutDuration {
+            final long duration;
+            final TimeUnit timeUnit;
+
+            TimeoutDuration(long duration, TimeUnit timeUnit) {
+
+                this.duration = duration;
+                this.timeUnit = timeUnit;
+            }
+
+        }
 
         @Override
         public U get() throws InterruptedException, ExecutionException {
+            try {
+                return get(Optional.empty());
+            } catch (TimeoutException e) {
+                return null;
+            }
+        }
+
+        @Override
+        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+
+            return get(Optional.of(new TimeoutDuration(timeout, unit)));
+        }
+
+        private U get(Optional<TimeoutDuration> timeout)
+                throws TimeoutException, InterruptedException, ExecutionException {
             if (cancelled) {
                 throw new CancellationException();
             }
 
             if (!completed) {
-                completer.triggerFutureCompletion();
-
+                if (timeout.isPresent()) {
+                    ExecutorService executor = Executors.newSingleThreadExecutor();
+                    try {
+                        Future<?> submittedTask = executor.submit(() -> {
+                            completer.triggerFutureCompletion();
+                        });
+                        submittedTask.get(timeout.get().duration, timeout.get().timeUnit);
+                    } catch (TimeoutException e) {
+                        throw new TimeoutException("The request has timed out");
```

Comment:
This catch block can go away–I think we can just let that TimeoutException propagate. There's no reason to rethrow I think.